src/structures/Match.js

```diff
@@ -0,0 +1,354 @@
+const Client = require('../client/Client'); // eslint-disable-line no-unused-vars
+const {MMTypes, MatchmakingGroup, MatchStatus} = require('../util/Constants'); // eslint-disable-line no-unused-vars
+const RoyalTeams = require('../data/RoyalTeams.json');
+
+/**
+ * The match
+ */
+class Match {
+    constructor(client, data){
+        /**
+         * The client instance.
+         * @type {Client}
+         */
+        this.client = client;
+
+        /**
+         * The data
+         * @type {Object}
+         * @private
+         */
+        this._data = data;
+    }
+
+    /**
+     * The match ID
+     * @type {number}
+     */
+    get id(){
+        return this._data.id;
+    }
+
+    /**
+     * The match live ID
+     * @type {string}
+     */
+    get liveId(){
+        return this._data.lid;
+    }
+
+    /**
+     * The match name
+     * @type {string}
+     */
+    get name(){
+        return this._data.name;
+    }
+
+    /**
+     * The match type
+     * @type {?MatchmakingGroup}
+     */
+    get type(){
+        if (this.name.includes("3v3")) return MMTypes[2];
+        else if (this.name.includes("royal")) return MMTypes[3];
+        else return null;
+    }
+
+    /**
+     * The match group
+     * @type {string}
+     * @private
+     */
+    get group(){
+        return this._data.group;
+    }
+
+    /**
+     * The match start date
+     * @type {Date}
+     */
+    get startDate(){
+        return new Date(this._data.startdate * 1000);
+    }
+
+    /**
+     * The match end date
+     * @type {Date}
+     */
+    get endDate(){
+        return new Date(this._data.enddate * 1000);
+    }
+
+    /**
+     * The match score direction
+     * @type {string}
+     * @private
+     */
+    get scoreDirection(){
+        return this._data.scoredirection;
```

Comment:
Is this a French/English translation thing? Not sure what a "score direction" is 🤔

Reply:
this cames from the API, but seems to return the same value on every matches, so I set up that to private